src/minidriver/minidriver.c

```diff
@@ -1490,16 +1490,16 @@ md_set_cmapfile(PCARD_DATA pCardData, struct md_file *file)
 		 */
 		cont->size_key_exchange = cont->size_sign = 0;
 		if (key_obj->type == SC_PKCS15_TYPE_PRKEY_RSA) {
-			if (prkey_info->usage & USAGE_ANY_DECIPHER)
+			if ((prkey_info->usage & USAGE_ANY_DECIPHER) == USAGE_ANY_DECIPHER)
 				cont->size_key_exchange = prkey_info->modulus_length;
-			else if (prkey_info->usage & USAGE_ANY_SIGN)
+			else if ((prkey_info->usage & USAGE_ANY_SIGN) == USAGE_ANY_SIGN)
 				cont->size_sign = prkey_info->modulus_length;
 			else
 				cont->size_key_exchange = prkey_info->modulus_length;
 		} else if (key_obj->type == SC_PKCS15_TYPE_PRKEY_EC) {
-			if (prkey_info->usage & USAGE_ANY_AGREEMENT)
+			if ((prkey_info->usage & USAGE_ANY_AGREEMENT) == USAGE_ANY_AGREEMENT)
 				cont->size_key_exchange = prkey_info->field_length;
-			else if (prkey_info->usage & USAGE_ANY_SIGN)
+			else if ((prkey_info->usage & USAGE_ANY_SIGN) == USAGE_ANY_SIGN)
```

Comment:
Our JPKI card has two keys:

~~~
% pkcs15-tool -D
Private RSA Key [User Authentication Key]
        Object Flags   : [0x1], private
        Usage          : [0x4], sign

Private RSA Key [Digital Signature Key]
        Object Flags   : [0x1], private
        Usage          : [0x204], sign, nonRepudiation
~~~

The first key for authentication such as TLS client auth, and second one for signature.
Also other national cards are roughly  similar.
But minidriver would handle both key as signature(set size_sign and not set size_key_exchange)
Is this expected behavior?
Thanks for your comment.